presto-hive/src/main/java/com/facebook/presto/hive/HiveBasicStatistics.java

```diff
@@ -78,6 +80,20 @@ public OptionalLong getOnDiskDataSizeInBytes()
         return onDiskDataSizeInBytes;
     }
 
+    public HiveBasicStatistics applyUpdate(HiveBasicStatistics statisticsUpdate)
+    {
+        return new HiveBasicStatistics(
+                firstPresent(HiveBasicStatistics::getFileCount, statisticsUpdate, this),
+                firstPresent(HiveBasicStatistics::getRowCount, statisticsUpdate, this),
+                firstPresent(HiveBasicStatistics::getInMemoryDataSizeInBytes, statisticsUpdate, this),
+                firstPresent(HiveBasicStatistics::getOnDiskDataSizeInBytes, statisticsUpdate, this));
+    }
+
+    private OptionalLong firstPresent(Function<HiveBasicStatistics, OptionalLong> getter, HiveBasicStatistics first, HiveBasicStatistics second)
```

Comment:
Keep it `private static` in `SemiTransactionalHiveMetastore`.

Also i think the `private OptionalLong firstPresent(OptionalLong first, OptionalLong second)` is more readable